src/commonControls/MultiValueListControl/MultiValueListControl.ts

```diff
@@ -0,0 +1,1382 @@
+            handle: this.handleSelectDoneByTouch,
+        },
+    ];
+    // tsDoc - see Control
+    async canHandle(input: ControlInput): Promise<boolean> {
+        const stdHandlers = this.standardInputHandlers;
+        const customHandlers = this.props.inputHandling.customHandlingFuncs ?? [];
+
+        const matches = [];
+        for (const handler of stdHandlers.concat(customHandlers)) {
+            if (await handler.canHandle.call(this, input)) {
+                matches.push(handler);
+            }
+        }
+
+        if (matches.length > 1) {
+            log.error(
+                `More than one handler matched. Handlers in a single control should be mutually exclusive. ` +
+                    `Defaulting to the first. handlers: ${JSON.stringify(matches.map((x) => x.name))}`,
+            );
+        }
+
+        if (matches.length >= 1) {
+            this.handleFunc = matches[0].handle.bind(this);
+            return true;
+        } else {
+            return false;
+        }
+    }
+
+    // tsDoc - see Control
+    async handle(input: ControlInput, resultBuilder: ControlResultBuilder): Promise<void> {
+        if (this.handleFunc === undefined) {
+            log.error(
+                'MultiValueListControl: handle called but no clause matched. are canHandle/handle out of sync?',
+            );
+            const intent: Intent = (input.request as IntentRequest).intent;
+            throw new Error(`${intent.name} can not be handled by ${this.constructor.name}.`);
+        }
+
+        this.props.required = true;
+        await this.handleFunc(input, resultBuilder);
+        if (resultBuilder.hasInitiativeAct() !== true && (await this.canTakeInitiative(input)) === true) {
+            await this.takeInitiative(input, resultBuilder);
+        }
+    }
+
+    private isAddWithValue(input: ControlInput): boolean {
+        try {
+            okIf(InputUtil.isIntent(input, MultiValueControlIntent.intentName(this.props.slotType)));
+            const { feedback, action, target, values, valueType } = unpackMultiValueControlIntent(
+                (input.request as IntentRequest).intent,
+            );
+            okIf(InputUtil.targetIsMatchOrUndefined(target, this.props.interactionModel.targets));
+            okIf(InputUtil.valueTypeMatch(valueType, this.getSlotTypes()));
+            okIf(InputUtil.valueStrDefined(values));
+            okIf(InputUtil.feedbackIsMatchOrUndefined(feedback, this.getFeedBackTypes()));
+            okIf(InputUtil.actionIsMatchOrUndefined(action, this.props.interactionModel.actions.add));
+            return true;
+        } catch (e) {
+            return falseIfGuardFailed(e);
+        }
+    }
+
+    private async handleAddWithValue(
+        input: ControlInput,
+        resultBuilder: ControlResultBuilder,
+    ): Promise<void> {
+        const slotValues = InputUtil.getMultiValueResolution(input);
+        let values = this.generateSlotValues(slotValues);
```

Comment:
generateSlotValue is an odd name.  getSlotValues?

Reply:
Updated to `getSlotValues()`